geode-core/src/test/java/org/apache/geode/management/internal/web/http/support/HttpRequesterTest.java

```diff
@@ -0,0 +1,154 @@
+
+  }
+
+  @Test
+  public void extractResponseOfJsonString() throws Exception {
+    String responseString = "my response";
+    requester = new HttpRequester();
+    response =
+        new MockClientHttpResponse(IOUtils.toInputStream(responseString, "UTF-8"), HttpStatus.OK);
+
+    response.getHeaders().add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+    Object result = requester.extractResponse(response);
+    assertThat(result).isEqualTo(responseString);
+  }
+
+  @Test
+  public void extractResponseOfFileDownload() throws Exception {
+    File responseFile = temporaryFolder.newFile();
+    FileUtils.writeStringToFile(responseFile, "some file contents", "UTF-8");
+    requester = new HttpRequester();
+    response = new MockClientHttpResponse(new FileInputStream(responseFile), HttpStatus.OK);
+    response.getHeaders().add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_OCTET_STREAM_VALUE);
+    Object result = requester.extractResponse(response);
+    Path fileResult = (Path) result;
+    assertThat(fileResult).hasSameContentAs(responseFile.toPath());
+  }
+
+  @Test
+  public void createURI() throws Exception {
+    requester = new HttpRequester();
+    assertThat(requester.createURI("http://test.org", "abc").toString())
+        .isEqualTo("http://test.org/abc");
+    assertThat(requester.createURI("http://test.org", "abc", "key", "value").toString())
+        .isEqualTo("http://test.org/abc?key=value");
+
+    assertThat(requester.createURI("http://test.org", "abc", "a-b", "c d").toString())
+        .isEqualTo("http://test.org/abc?a-b=c%20d");
+
+    assertThatThrownBy(() -> requester.createURI("http://test.org", "abc", "key"))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void get() throws Exception {
+    requester = spy(new HttpRequester(securityProps, restTemplate));
+    String result = requester.get(uri, String.class);
+
+    assertThat(result).isEqualTo("done");
+    verify(requester).exchange(uri, HttpMethod.GET, null, null, String.class);
+
+    verifyHeaderIsUpdated();
+  }
+
+  @Test
+  public void post() throws Exception {
+    requester = spy(new HttpRequester(securityProps, restTemplate));
+    String result =
+        requester.post(uri, MediaType.APPLICATION_FORM_URLENCODED, "myData", String.class);
+
+    assertThat(result).isEqualTo("done");
+    verify(requester).exchange(uri, HttpMethod.POST, MediaType.APPLICATION_FORM_URLENCODED,
+        "myData", String.class);
+
+    verifyHeaderIsUpdated();
+  }
+
+  private void verifyHeaderIsUpdated() {
+    ArgumentCaptor<HttpHeaders> headerCaptor = ArgumentCaptor.forClass(HttpHeaders.class);
+    verify(requester).addHeaderValues(headerCaptor.capture());
+
+    HttpHeaders headers = headerCaptor.getValue();
+    assertThat(headers.get("user").get(0)).isEqualTo("me");
+    assertThat(headers.get("password").get(0)).isEqualTo("secret");
+  }
+
+  @Test
+  public void authenticateError() throws Exception {
+
```

Comment:
Can you add a test or remove this method if it's not relevant?

Reply:
deleted.